--- 
title: "Characterisation of model species interactome available from primary molecular interaction databases: for paper"
author: "Vitalii Kleshchevnikov"
date: "23 November 2017"
output:
html_document: default
--- 

```{r setup, include=FALSE} 
knitr::opts_chunk$set(echo = FALSE, results = "hide", warning = FALSE, message = FALSE, fig.keep = "all",eval=T) 

packages = c("MItools", "data.table", "dplyr", "UniProt.ws", "PSICQUIC", "ggplot2", "clusterProfiler", "scales", "VennDiagram", "reshape2", "MASS", "RColorBrewer", "multcomp")
if(mean(packages %in% names(installed.packages()[,"Package"])) != 1){
  packages_to_install = packages[!packages %in% names(installed.packages()[,"Package"])]
  # specifying mirror is necessary for some Linux systems
  install.packages(packages_to_install, dependencies = TRUE, repos = "http://mirrors.ebi.ac.uk/CRAN/")
  packages_to_install = packages[!packages %in% names(installed.packages()[,"Package"])]
  source("https://bioconductor.org/biocLite.R")
  biocLite(packages_to_install, dependencies = TRUE)
  # install my package
  devtools::install_github("vitkl/MItools", dependencies = TRUE)
  # install the packages I have modified (necessary to for GO enrichment)
  devtools::install_github("vitkl/clusterProfiler", dependencies = TRUE)
  devtools::install_github("vitkl/DOSE")
}

suppressPackageStartupMessages({
  library(data.table)
  library(dplyr)
  library(UniProt.ws)
  library(PSICQUIC)
  library(ggplot2)
  library(clusterProfiler)
  library(scales)
  library(VennDiagram)
  library(reshape2)
  library(MASS)
  library(RColorBrewer)
  library(multcomp)
  library(MItools)
})

source("multiplot.R")
source("GO_analyse_simplify_plot.R")


date = Sys.Date()
## Enter date manually to use older data downloaded on a specific date
date = as.Date("2017-11-22")
``` 

The report was published on `r Sys.Date()`. Data used in the report was available on `r date`.


### Computational analysis: code

All R code used in this analysis in deposited to a public Github repository: https://github.com/vitkl/imex_vs_uniprot

## SwissProt coverage

```{r IMEx_vs_Uniprot_processing, cache=F}
#### This chunk processes data
filename.summary.all = paste("./results/","proteome_vs_interactome_summary_",date,".txt", sep = "")
filename.summary.all2 = paste("./results/","whole_proteome_Uniprot_vs_IMEx_vs_protein.exist_summary_",date,".txt", sep = "")
filename.summary.all3 = paste("./results/","whole_proteome_Uniprot_vs_IntAct_vs_protein.exist_summary_",date,".txt", sep = "")
# if the analysis has been done - don't do it again
if(!file.exists(filename.summary.all)){
  # download are read all IntAct data from FTP  
  IntActFTP = MItools::loadIntActFTP("./Data/")
  
  SPECIES_NAME = c("Homo sapiens", "Mus musculus", "strain ATCC 204508", "strain K12", "Drosophila melanogaster", "Caenorhabditis elegans", "Arabidopsis thaliana")
  
  ## ## Use all Uniprot if reviewed == 1, only Swissprot data if reviewed == 2, 
  ## ## TrEMBL data if reviewed == 3
  reviewed = c(1, 2)
  
  ## ## Distinguish between isoforms or use only generic Uniprot IDs: TRUE / FALSE?
  isoforms = c(TRUE, FALSE)
  
  ## missing_proteins = TRUE => proteins missing protein evidence are shown
  ## missing_proteins = FALSE => proteins for which protein evidence exists are shown
  missing_proteins = TRUE
  
  ## The code below queries databases, saves and processes results, 
  ## gives summary on how many interactors have Uniprot IDs or belong to the SPECIES_NAME
  ## generates table with 0 and 1 for the combination of SPECIES_NAME, reviewed, isoforms
  
  ## type in exact database names (the list below is the default for the function)
  databases <- c("IntAct", "MINT", "bhf-ucl", "MPIDB", "MatrixDB", 
                 "HPIDb","I2D-IMEx","InnateDB-IMEx", "MolCon", "UniProt", "MBInfo")
  
  source("intact_vs_uniprot.R")
  for (i in isoforms) {
    for (r in reviewed) {
      for (n in SPECIES_NAME) {
        intact_vs_uniprot(SPECIES_NAME = n, reviewed = r, isoforms = i, missing_proteins = TRUE, date = date, databases = databases, fullIntAct = IntActFTP)
        message(paste("- ",n," -  reviewed", r, " -  isoforms included", i))
      }
    }
  }
  
  ## The code below does overlap comparisons and saves summaries
  source("intact_vs_uniprot_overlap.R")
  source("SPECIES_NAME_TO_ID.R")
  for (r in reviewed) {
    for (i in isoforms) {
      for (n in SPECIES_NAME) {
        SPECIES_IDs = SPECIES_NAME_TO_ID(n)
        SPECIES_ID = SPECIES_IDs$SPECIES_ID
        intact_vs_uniprot_overlap(SPECIES_NAME = n, SPECIES_ID = SPECIES_ID, reviewed = r, isoforms = i, date)
        message(paste("- ",n," -  reviewed", r, " -  isoforms included", i))
      }
    }
  }
  
  ### combine the summaries for multiple species
  proteome_vs_interactome_summary.all = data.frame()
  whole_proteome_Uniprot_vs_IMEx_vs_protein.exist.all = data.frame()
  whole_proteome_Uniprot_vs_IntAct_vs_protein.exist.all = data.frame()
  for (r in reviewed) {
    for (i in isoforms) {
      for(n in SPECIES_NAME) {
        #=====================================
        filename.summary = paste("./summaries/","proteome_vs_interactome_summary_",n,"_reviewed_",r,"_isoforms_",i,"_",date,".txt", sep = "")
        if(!file.exists(filename.summary)) {
          proteome_vs_interactome_summary.all
        }
        if(file.exists(filename.summary)) {
          proteome_vs_interactome_summary_o = as.data.frame(read.delim(filename.summary, stringsAsFactors = F, quote = ""))
          proteome_vs_interactome_summary.all = rbind(proteome_vs_interactome_summary.all, proteome_vs_interactome_summary_o)
        }
        #=====================================
        filename.summary2 = paste("./summaries/","whole_proteome_Uniprot_vs_IMEx_vs_protein.exist_summary_",n,"_reviewed_",r,"_isoforms_",i,"_",date,".txt", sep = "")
        if(!file.exists(filename.summary2)) {
          whole_proteome_Uniprot_vs_IMEx_vs_protein.exist.all
        }
        if(file.exists(filename.summary2)) {
          whole_proteome_Uniprot_vs_IMEx_vs_protein.exist = as.data.frame(read.delim(filename.summary2, stringsAsFactors = F, quote = ""))
          whole_proteome_Uniprot_vs_IMEx_vs_protein.exist.all = rbind(whole_proteome_Uniprot_vs_IMEx_vs_protein.exist.all, whole_proteome_Uniprot_vs_IMEx_vs_protein.exist)
        }
        #=====================================
      }
    }
  }
  ## save all species summary
  filename.summary.all = paste("./results/","proteome_vs_interactome_summary_",date,".txt", sep = "")
  write.table(proteome_vs_interactome_summary.all, filename.summary.all, col.names=T,row.names=F,sep="\t",quote=F)
  filename.summary.all2 = paste("./results/","whole_proteome_Uniprot_vs_IMEx_vs_protein.exist_summary_",date,".txt", sep = "")
  write.table(whole_proteome_Uniprot_vs_IMEx_vs_protein.exist.all, filename.summary.all2, col.names=T,row.names=F,sep="\t",quote=F)
}

{
  if(file.exists(filename.summary.all)) proteome_vs_interactome_summary.all = fread(filename.summary.all, sep = "\t", stringsAsFactors = F, data.table = F)
}
```

```{r IMEx_vs_Uniprot_venndiagram, cache=F, fig.width=8, fig.height=8, fig.cap="The best coverage is observed for yeast, E.coli and human. Isoform coverage is limited, but still significant for human. Red circles represent proteins which have interacting partners curated in IntAct, blue circles represent proteins in SwissProt, taking into account isoforms (left) or canonical identifiers (right)."}
#### This chunk plots venn.diagram - Uniprot vs IMEx

# code to be modified for different graphs
SPECIES_NAME = c("Homo sapiens", "Mus musculus", "strain ATCC 204508", "strain K12", "Drosophila melanogaster", "Caenorhabditis elegans", "Arabidopsis thaliana")
reviewed_venn = 2                 # 1 or 2
IMExdatabase_sel = "IMEx"    # subset the results "X0469.IntAct." or "IMEx"
IMExdatabase_label = "IntAct" # show on the plot
### ======================================================================== ###
# non-modifyable code
proteome_vs_interactome_summary.all$species.name = gsub("strain ATCC 204508", "S. cerevisiae, strain S288c", proteome_vs_interactome_summary.all$species.name)
proteome_vs_interactome_summary.all$species.name = gsub("strain K12", "Escherichia coli, strain K12", proteome_vs_interactome_summary.all$species.name)
SPECIES_NAME = gsub("strain ATCC 204508", "S. cerevisiae, strain S288c", SPECIES_NAME)
SPECIES_NAME = gsub("strain K12", "Escherichia coli, strain K12", SPECIES_NAME)
proteome_vs_interactome_summary.all_isof = filter(proteome_vs_interactome_summary.all, reviewed == reviewed_venn & isoforms == TRUE & database == IMExdatabase_sel)
proteome_vs_interactome_summary.all_noisof = filter(proteome_vs_interactome_summary.all, reviewed == reviewed_venn & isoforms == FALSE & database == IMExdatabase_sel)

{
  source("double.venn.prot.R")
  grid.newpage()
  
  if(reviewed_venn == 2){plotname = paste0("Proteome (SwissProt) coverage by interaction databases(",IMExdatabase_label,")")
  databasename = "SwissProt"}
  if(reviewed_venn == 1){plotname = paste0("Proteome (all UniprotKB) coverage by interaction databases(",IMExdatabase_label,")")
  databasename = "UniprotKB"}
  
  pushViewport(viewport(layout=grid.layout(nrow = length(SPECIES_NAME)+2, ncol=4, widths = unit(c(2/7,2/7,2/7,1/7), "npc"), 
                                           heights = unit(c(1/((length(SPECIES_NAME)+1)*4),1/((length(SPECIES_NAME)+1)*4),rep(1/(length(SPECIES_NAME)+1/2),length(SPECIES_NAME))), "npc"))))
  
  pushViewport(viewport(layout.pos.col=2, layout.pos.row = 1))
  x =grid.text(plotname, x = unit(0.7, "npc"),y= unit(0.5, "npc"))
  popViewport()
  pushViewport(viewport(layout.pos.col=2, layout.pos.row = 2))
  x =grid.text("isoforms included", x = unit(0.5, "npc"),y= unit(0.5, "npc"))
  popViewport()
  pushViewport(viewport(layout.pos.col=3, layout.pos.row = 2))
  x =grid.text("isoforms excluded", x = unit(0.5, "npc"),y= unit(0.5, "npc"))
  popViewport()
  
  for (i in 1:length(SPECIES_NAME)) {
    
    pushViewport(viewport(layout.pos.col=1, layout.pos.row = i+2))
    x =grid.text(SPECIES_NAME[i], x = unit(0.5, "npc"),y= unit(0.5, "npc"))
    popViewport()
    
    pushViewport(viewport(layout.pos.col=2, layout.pos.row = i+2))
    venn = double.venn.prot(proteome_vs_interactome_summary.all_isof[i,], scaled = FALSE, IMExdatabase = IMExdatabase_label)
    popViewport()
    
    pushViewport(viewport(layout.pos.col=3, layout.pos.row = i+2))
    venn = double.venn.prot(proteome_vs_interactome_summary.all_noisof[i,], scaled = FALSE, IMExdatabase = IMExdatabase_label)
    popViewport()
  }
  popViewport()
}

human_isoform = proteome_vs_interactome_summary.all_isof$whole.proteome..Uniprot[proteome_vs_interactome_summary.all_isof["species.name"]=="Homo sapiens"] - proteome_vs_interactome_summary.all_noisof$whole.proteome..Uniprot[proteome_vs_interactome_summary.all_isof["species.name"]=="Homo sapiens"]

human_isoform_with_interactions = proteome_vs_interactome_summary.all_isof$whole.proteome..interactome.available[proteome_vs_interactome_summary.all_isof["species.name"]=="Homo sapiens"] - proteome_vs_interactome_summary.all_noisof$whole.proteome..interactome.available[proteome_vs_interactome_summary.all_isof["species.name"]=="Homo sapiens"] + (proteome_vs_interactome_summary.all_isof$interactome..but.not.in.Uniprot[proteome_vs_interactome_summary.all_isof["species.name"]=="Homo sapiens"] - proteome_vs_interactome_summary.all_noisof$interactome..but.not.in.Uniprot[proteome_vs_interactome_summary.all_isof["species.name"]=="Homo sapiens"])
```

## trEMBL coverage

```{r Supplementary1_IMEx_vs_Uniprot_venndiagram_all_Uniprot, cache=F, fig.width=8, fig.height=8, fig.cap="Supplementary figure 1. The overlap between proteins with known interactions (interactome) and all proteins included in UniprotKB (proteome). Red circles represent proteins which have interacting partners curated in IntAct, blue circles represent proteins in SwissProt, taking into account isoforms (left) or canonical identifiers (right)."}
#### This chunk plots venn.diagram - Uniprot vs IMEx

filename.summary.all = paste("./results/","proteome_vs_interactome_summary_",date,".txt", sep = "")
{
  if(file.exists(filename.summary.all)) proteome_vs_interactome_summary.all = fread(filename.summary.all, sep = "\t", stringsAsFactors = F, data.table = F)
}
## Plotting with venn.diagram - Uniprot vs IMEx (or IntAct)
# code to be modified for different graphs
SPECIES_NAME = c("Homo sapiens", "Mus musculus", "strain ATCC 204508", "strain K12", "Drosophila melanogaster", "Caenorhabditis elegans", "Arabidopsis thaliana")
reviewed_venn = 1                 # 1 or 2
IMExdatabase = "IMEx"    # "X0469.IntAct." or "IMEx"
### ======================================================================== ###
# non-modifyable code
proteome_vs_interactome_summary.all$species.name = gsub("strain ATCC 204508", "S. cerevisiae, strain S288c", proteome_vs_interactome_summary.all$species.name)
proteome_vs_interactome_summary.all$species.name = gsub("strain K12", "Escherichia coli, strain K12", proteome_vs_interactome_summary.all$species.name)
SPECIES_NAME = gsub("strain ATCC 204508", "S. cerevisiae, strain S288c", SPECIES_NAME)
SPECIES_NAME = gsub("strain K12", "Escherichia coli, strain K12", SPECIES_NAME)
proteome_vs_interactome_summary.all_isof = filter(proteome_vs_interactome_summary.all, reviewed == reviewed_venn)
proteome_vs_interactome_summary.all_isof = filter(proteome_vs_interactome_summary.all_isof, isoforms == TRUE)
proteome_vs_interactome_summary.all_isof = filter(proteome_vs_interactome_summary.all_isof, database == IMExdatabase)
proteome_vs_interactome_summary.all_noisof = filter(proteome_vs_interactome_summary.all, reviewed == reviewed_venn)
proteome_vs_interactome_summary.all_noisof = filter(proteome_vs_interactome_summary.all_noisof, isoforms == FALSE)
proteome_vs_interactome_summary.all_noisof = filter(proteome_vs_interactome_summary.all_noisof, database == IMExdatabase)

{
  grid.newpage()
  
  if(reviewed_venn == 2){plotname = paste0("Proteome (SwissProt) coverage by interaction databases(",IMExdatabase,")")
  databasename = "SwissProt"}
  if(reviewed_venn == 1){plotname = paste0("Proteome (all UniprotKB) coverage by interaction databases(",IMExdatabase,")")
  databasename = "UniprotKB"}
  
  pushViewport(viewport(layout=grid.layout(nrow = length(SPECIES_NAME)+2, ncol=4, widths = unit(c(2/7,2/7,2/7,1/7), "npc"), 
                                           heights = unit(c(1/((length(SPECIES_NAME)+1)*4),1/((length(SPECIES_NAME)+1)*4),rep(1/(length(SPECIES_NAME)+1/2),length(SPECIES_NAME))), "npc"))))
  
  pushViewport(viewport(layout.pos.col=2, layout.pos.row = 1))
  x =grid.text(plotname, x = unit(0.7, "npc"),y= unit(0.5, "npc"))
  popViewport()
  pushViewport(viewport(layout.pos.col=2, layout.pos.row = 2))
  x =grid.text("isoforms included", x = unit(0.5, "npc"),y= unit(0.5, "npc"))
  popViewport()
  pushViewport(viewport(layout.pos.col=3, layout.pos.row = 2))
  x =grid.text("isoforms excluded", x = unit(0.5, "npc"),y= unit(0.5, "npc"))
  popViewport()
  
  for (i in 1:length(SPECIES_NAME)) {
    
    pushViewport(viewport(layout.pos.col=1, layout.pos.row = i+2))
    x =grid.text(SPECIES_NAME[i], x = unit(0.5, "npc"),y= unit(0.5, "npc"))
    popViewport()
    
    pushViewport(viewport(layout.pos.col=2, layout.pos.row = i+2))
    source("double.venn.prot.R")
    venn = double.venn.prot(proteome_vs_interactome_summary.all_isof[i,], scaled = FALSE, IMExdatabase = IMExdatabase)
    popViewport()
    
    pushViewport(viewport(layout.pos.col=3, layout.pos.row = i+2))
    source("double.venn.prot.R")
    venn = double.venn.prot(proteome_vs_interactome_summary.all_noisof[i,], scaled = FALSE, IMExdatabase = IMExdatabase)
    popViewport()
  }
  popViewport()
}
```

## 3. BioGRID vs IntAct overlap

```{r biogrid_vs_IMEx_vs_Uniprot_processing}
#### This chunk processes the data
filename.summary.all_biog = paste("./results/","whole_proteome_Uniprot_vs_IMEx_vs_BioGRID_from_Mentha_summary_",date,".txt", sep = "")
if(!file.exists(filename.summary.all_biog)){
  
  ## Enter SPECIES_NAME 
  SPECIES_NAME = c("Homo sapiens", "Mus musculus", "strain ATCC 204508", "Drosophila melanogaster", "Caenorhabditis elegans", "Arabidopsis thaliana")
  ## !! no "E.coli strain K12" in BioGRID from Mentha
  ## ## Use all Uniprot if reviewed == 1, only Swissprot data if reviewed == 2, 
  ## ## TrEMBL data if reviewed == 3
  reviewed = c(1, 2)
  ## ## Distinguish between isoforms or use only generic Uniprot IDs: TRUE / FALSE?
  isoforms = c(FALSE) # not possible to distinguish isoforms for BioGRID
  
  #============================================================================#
  source("biogrid_from_mentha_vs_proteome_vs_imex.R")
  for (r in reviewed) {
    for (i in isoforms) {
      for (n in SPECIES_NAME) {
        biogrid_from_mentha_vs_proteome_vs_imex(SPECIES_NAME = n, reviewed = r, isoforms = i, date = date)
      }
    }
  }
  
  whole_proteome_Uniprot_vs_IMEx_vs_BioGRID_from_Mentha = data.frame()
  reference_proteome_Uniprot_vs_IMEx_vs_BioGRID_from_Mentha = data.frame()
  whole_proteome_Uniprot_vs_IntAct_vs_BioGRID_from_Mentha = data.frame()
  source("SPECIES_NAME_TO_ID.R")
  for (r in reviewed) {
    for (i in isoforms) {
      for (n in SPECIES_NAME) {
        SPECIES_IDs = SPECIES_NAME_TO_ID(n)
        SPECIES_ID = SPECIES_IDs$SPECIES_ID
        filename_vs_3 = paste("./analysis/","proteome_vs_interactome_vs_BioGRID_f_", SPECIES_ID,"_reviewed_",reviewed = r,"_isoforms_",isoforms = i,"_", date,".txt", sep = "")
        biogrid_from_mentha_vs_proteome_vs_imex_f = as.data.frame(read.delim(filename_vs_3, stringsAsFactors = F, quote = ""))
        
        #============================================================================#
        ## Calculating overlaps
        
        source("A_vs_B_vs_C_overlap.R")
        whole_proteome_Uniprot_vs_IMEx_vs_BioGRID_from_Mentha_temp = A_vs_B_vs_C_overlap(biogrid_from_mentha_vs_proteome_vs_imex_f, 
                                                                                         A = "whole_proteome_Uniprot", 
                                                                                         B = "IMEx", 
                                                                                         C = "BioGRID_from_Mentha", 
                                                                                         SPECIES_NAME = n, reviewed = r, isoforms = i)
        whole_proteome_Uniprot_vs_IMEx_vs_BioGRID_from_Mentha = rbind(whole_proteome_Uniprot_vs_IMEx_vs_BioGRID_from_Mentha, whole_proteome_Uniprot_vs_IMEx_vs_BioGRID_from_Mentha_temp)
        
        reference_proteome_Uniprot_vs_IMEx_vs_BioGRID_from_Mentha_temp = A_vs_B_vs_C_overlap(biogrid_from_mentha_vs_proteome_vs_imex_f, 
                                                                                             A = "reference_proteome_Uniprot", 
                                                                                             B = "IMEx", 
                                                                                             C = "BioGRID_from_Mentha", 
                                                                                             SPECIES_NAME = n, reviewed = r, isoforms = i)
        reference_proteome_Uniprot_vs_IMEx_vs_BioGRID_from_Mentha = rbind(reference_proteome_Uniprot_vs_IMEx_vs_BioGRID_from_Mentha, reference_proteome_Uniprot_vs_IMEx_vs_BioGRID_from_Mentha_temp)
      }
    }
  }
  write.table(whole_proteome_Uniprot_vs_IMEx_vs_BioGRID_from_Mentha, filename.summary.all_biog, col.names=T,row.names=F,sep="\t",quote=F)
}

{
  if(file.exists(filename.summary.all_biog)) whole_proteome_Uniprot_vs_IMEx_vs_BioGRID_from_Mentha = fread(filename.summary.all_biog, sep = "\t", stringsAsFactors = F, data.table = F)
}
```

```{r biogrid_vs_IMEx_vs_Uniprot_venndiagram, cache=F, fig.width=8, fig.height=8, fig.cap="BioGRID and IMEx share a considerable fraction of proteins with interaction data. Red circles represent proteins which have interacting partners curated in IntAct, blue circles represent proteins in SwissProt, green circles represent proteins which have interacting partners curated in BioGRID, taking into account all UniProtKB (left) or only manually reviewed SwissProt (right) proteins, only canonical identifiers considered." }
#### This chunk plots venn.diagram: the overlap between Uniprot, IMEx, BioGRID

SPECIES_NAME = c("Homo sapiens", "Mus musculus", "strain ATCC 204508", "Drosophila melanogaster", "Caenorhabditis elegans", "Arabidopsis thaliana")
IMExdatabase = "IMEx"    # "X0469.IntAct." or "IMEx"
IMExdatabase_label = "IntAct" # show on the plot
### ======================================================================== ###
# non-modifyable code
if(IMExdatabase == "IMEx"){
  whole_proteome_Uniprot_vs_IMEx_vs_BioGRID_from_Mentha$species_name = gsub("strain ATCC 204508", "S. cerevisiae, strain S288c", whole_proteome_Uniprot_vs_IMEx_vs_BioGRID_from_Mentha$species_name)
  proteome_vs_interactome_summary.all_isof = filter(whole_proteome_Uniprot_vs_IMEx_vs_BioGRID_from_Mentha, reviewed == 1)
  proteome_vs_interactome_summary.all_isof = filter(proteome_vs_interactome_summary.all_isof, isoforms == FALSE)
  proteome_vs_interactome_summary.all_noisof = filter(whole_proteome_Uniprot_vs_IMEx_vs_BioGRID_from_Mentha, reviewed == 2)
  proteome_vs_interactome_summary.all_noisof = filter(proteome_vs_interactome_summary.all_noisof, isoforms == FALSE)
}
if(IMExdatabase == "X0469.IntAct."){
  whole_proteome_Uniprot_vs_IntAct_vs_BioGRID_from_Mentha$species_name = gsub("strain ATCC 204508", "S. cerevisiae, strain S288c", whole_proteome_Uniprot_vs_IntAct_vs_BioGRID_from_Mentha$species_name)
  proteome_vs_interactome_summary.all_isof = filter(whole_proteome_Uniprot_vs_IntAct_vs_BioGRID_from_Mentha, reviewed == 1)
  proteome_vs_interactome_summary.all_isof = filter(proteome_vs_interactome_summary.all_isof, isoforms == FALSE)
  proteome_vs_interactome_summary.all_noisof = filter(whole_proteome_Uniprot_vs_IntAct_vs_BioGRID_from_Mentha, reviewed == 2)
  proteome_vs_interactome_summary.all_noisof = filter(proteome_vs_interactome_summary.all_noisof, isoforms == FALSE)
}
{
  grid.newpage()
  
  plotname = paste0("Proteome coverage by interaction databases, overlap between ", IMExdatabase_label," and BioGRID")
  
  pushViewport(viewport(layout=grid.layout(nrow = length(SPECIES_NAME)+2, ncol=4, widths = unit(c(2/7,2/7,2/7,1/7), "npc"), 
                                           heights = unit(c(1/((length(SPECIES_NAME)+1)*4),1/((length(SPECIES_NAME)+1)*4),rep(1/(length(SPECIES_NAME)+1/2),length(SPECIES_NAME))), "npc"))))
  
  pushViewport(viewport(layout.pos.col=2, layout.pos.row = 1))
  x =grid.text(plotname, x = unit(0.7, "npc"),y= unit(0.5, "npc"))
  popViewport()
  pushViewport(viewport(layout.pos.col=2, layout.pos.row = 2))
  x =grid.text("UniprotKB, isoforms excluded", x = unit(0.5, "npc"),y= unit(0.5, "npc"))
  popViewport()
  pushViewport(viewport(layout.pos.col=3, layout.pos.row = 2))
  x =grid.text("SwissProt, isoforms excluded", x = unit(0.5, "npc"),y= unit(0.5, "npc"))
  popViewport()
  
  for (i in 1:length(SPECIES_NAME)) {
    
    pushViewport(viewport(layout.pos.col=1, layout.pos.row = i+2))
    x =grid.text(SPECIES_NAME[i], x = unit(0.5, "npc"),y= unit(0.5, "npc"))
    popViewport()
    
    pushViewport(viewport(layout.pos.col=2, layout.pos.row = i+2))
    source("triple.venn.prot.R")
    venn = triple.venn.prot(proteome_vs_interactome_summary.all_isof[i,], scaled = TRUE)
    popViewport()
    
    pushViewport(viewport(layout.pos.col=3, layout.pos.row = i+2))
    source("triple.venn.prot.R")
    venn = triple.venn.prot(proteome_vs_interactome_summary.all_noisof[i,], scaled = TRUE)
    popViewport()
  }
  popViewport()
}
```

## 3. Bar plot for interspecies interactions: IntAct

```{r IMEx_vs_Uniprot_N_Uniprot_Species, cache=F, fig.width=8, fig.height=8, fig.cap="Mouse and human proteins are commonly combined for interaction experiments. Red bar shows the number of proteins from a given species, blue bar shows the number of proteins from another species. Red group proteins form interactions both with red and blue group proteins, blue group proteins form interactions only with red group proteins" }
#### This chunk plots using ggplot
SPECIES_NAME = c("Homo sapiens", "Mus musculus", "strain ATCC 204508", "strain K12", "Drosophila melanogaster", "Caenorhabditis elegans", "Arabidopsis thaliana")
isoforms = FALSE
### combine the summaries for multiple species of how many interactors have non-uniprot and non-given-species identifiers
interactome_identifiers_summary.all = data.frame()
for (i in isoforms) {
  for(n in SPECIES_NAME) {
    filename.summary = paste("./summaries/","uniprotKB_IDs_and_",n,"_interactors_summary_isoforms_",i,"_",date,".txt", sep = "")
    if(!file.exists(filename.summary)) {
      interactome_identifiers_summary.all
    }
    if(file.exists(filename.summary)) {
      interactome_identifiers_summary.o = as.data.frame(fread(filename.summary, stringsAsFactors = F, data.table = F))
      interactome_identifiers_summary.all = rbind(interactome_identifiers_summary.all, interactome_identifiers_summary.o)
    }
  }
}
interactome_identifiers_summary.all = unique(interactome_identifiers_summary.all)
## Plotting the number of interactors which have UniprotKB identifiers and belong to the species queried
interactome_identifiers_summary.all.s = interactome_identifiers_summary.all[,c(3,4,5,6,7)]
## Transform table for plotting with ggplot2
interactome_identifiers_summary.all.s.m= melt(data = interactome_identifiers_summary.all.s,
                                              id.vars = c("SPECIES_NAME", "SPECIES_ID"),
                                              variable.name = "decription",
                                              value.name = "number")
## rename yeast and E.coli
interactome_identifiers_summary.all.s.m$SPECIES_NAME = gsub("strain ATCC 204508", "S. cerevisiae, strain S288c", interactome_identifiers_summary.all.s.m$SPECIES_NAME)
interactome_identifiers_summary.all.s.m$SPECIES_NAME = gsub("strain K12", "E. coli, strain K12", interactome_identifiers_summary.all.s.m$SPECIES_NAME)
interactome_identifiers_summary.all.s.m$SPECIES_NAME = gsub("Caenorhabditis elegans", "C. elegans", interactome_identifiers_summary.all.s.m$SPECIES_NAME)
interactome_identifiers_summary.all.s.m$SPECIES_NAME = gsub("Drosophila melanogaster", "D. melanogaster", interactome_identifiers_summary.all.s.m$SPECIES_NAME)

# filter out non-uniprot identifiers to make picture clearer
interactome_identifiers_summary.all.s.m$decription = gsub("\\.", " ", interactome_identifiers_summary.all.s.m$decription)
ind = interactome_identifiers_summary.all.s.m$decription != "interacting partners with the other identifier"
interactome_identifiers_summary.all.s.m = interactome_identifiers_summary.all.s.m[ind,]

########### PLOT
# plot how many interactors have non-uniprot and non-given-species identifiers
ggplot(interactome_identifiers_summary.all.s.m, aes(x=SPECIES_NAME, y=number, fill=decription)) + geom_bar(width = 0.9, stat = "identity", position = "stack") + geom_label(aes(label=number), position = "stack", size = 4, label.padding = unit(0.08, "lines")) +
  ggtitle("The number of interacting partners which have UniprotKB identifiers \n and belong to the species of interest") + theme(axis.text.x=element_text(angle = 20), axis.title.x = element_text(size = 0)) + ylab("number of proteins")
# save plot
# filename=paste("./results/", "interactome_identifiers_summary_plot_SMALL",date,".png", sep = "")
# ggsave(filename, proteome_vs_interactome_plot, width = 12, height = 12)
```


```{r dowlnload_all_IntAct, cache=F, fig.width=8}
#### This chunk processes the data for looking at bias in mass and GO molecular function
# IntAct FTP contains the following databases = c("IntAct", "MINT", "bhf-ucl", "MPIDB", "MatrixDB", 
#               "HPIDb","I2D-IMEx","InnateDB-IMEx", "MolCon", "UniProt", "MBInfo")
# this will take time and data traffic ~ 2.5GB

url_intact = "ftp://ftp.ebi.ac.uk/pub/databases/intact/current/psimitab/intact.txt"
filename_intact = paste0("./Data/IntAct_MITAB27_release_", format(Sys.Date(), "%m-%Y.txt"))
if(!file.exists(filename_intact)) {
  downloader::download(url_intact, filename_intact)
}
# Read all IntAct
all.IntAct = fread(filename_intact)
# changing column names to data.table-compatible format
{
  (colnames_IntAct = colnames(all.IntAct))
  colnames(all.IntAct) = gsub(" ","_",colnames(all.IntAct))
  colnames(all.IntAct) = gsub("\\(|\\)","",colnames(all.IntAct))
  colnames(all.IntAct) = gsub("#","",colnames(all.IntAct))
}
# cleaning Taxid "taxid:9606(human)|taxid:9606(Homo sapiens)" to 9606
all.IntAct = all.IntAct[,.(IDs_interactor_A, IDs_interactor_B, Experimental_roles_interactor_A, Experimental_roles_interactor_B,  Taxid_interactor_A, Taxid_interactor_B, Host_organisms, Publication_Identifiers, Confidence_values, Interaction_identifiers, Expansion_methods)]
{
  all.IntAct[, Taxid_interactor_A := gsub("taxid:|\\(.*$","",Taxid_interactor_A)]
  all.IntAct[, Taxid_interactor_B := gsub("taxid:|\\(.*$","",Taxid_interactor_B)]
  all.IntAct[, Host_organisms := gsub("taxid:|\\(.*$","",Host_organisms)]
  # saving identifier types and cleaning interactor ids
  all.IntAct[, interactor_IDs_databases_A := gsub(":.*$","",IDs_interactor_A)]
  all.IntAct[, interactor_IDs_databases_B := gsub(":.*$","",IDs_interactor_B)]
  all.IntAct[, IDs_interactor_A := gsub("^.*:","",IDs_interactor_A)]
  all.IntAct[, IDs_interactor_B := gsub("^.*:","",IDs_interactor_B)]
  # cleaning other information
  all.IntAct[, bait_prey_status_A := gsub("^.*\\(|\\)","",Experimental_roles_interactor_A)]
  all.IntAct[, bait_prey_status_B := gsub("^.*\\(|\\)","",Experimental_roles_interactor_B)]
  all.IntAct[, Publication_Identifiers := gsub("^.*pubmed:|\\|.*$","",Publication_Identifiers)]
  all.IntAct[, Confidence_values := gsub("^intact-miscore:","",Confidence_values)]
  all.IntAct[, Confidence_values := gsub("-","NA",Confidence_values)]
  all.IntAct[, Confidence_values := as.numeric(Confidence_values)]
  all.IntAct[, Interaction_identifiers := unlist(gsubfn::strapplyc(Interaction_identifiers,"EBI-[[:digit:]]+",simplify = T)), by =Interaction_identifiers]
}

# checking mixed mouse / human interactions
{
  the_host_for_mouse_human_pairs = all.IntAct[((Taxid_interactor_A == "9606" & Taxid_interactor_B == "10090")  | 
                                                 (Taxid_interactor_A == "10090" & Taxid_interactor_B == "9606")) &
                                                interactor_IDs_databases_A == "uniprotkb" & interactor_IDs_databases_B == "uniprotkb"
                                              , table(Host_organisms)]
  the_host_for_mouse_human_pairs_mouse_as_bait = all.IntAct[((Taxid_interactor_A == "9606" & Taxid_interactor_B == "10090")  | 
                                                               (Taxid_interactor_A == "10090" & Taxid_interactor_B == "9606")) &
                                                              interactor_IDs_databases_A == "uniprotkb" & interactor_IDs_databases_B == "uniprotkb" &
                                                              ((bait_prey_status_A == "prey" & bait_prey_status_B == "bait")  | 
                                                                 (bait_prey_status_A == "bait" & bait_prey_status_B == "prey"))
                                                            , table(Host_organisms)]
  mouse_as_bait = all.IntAct[((Taxid_interactor_A == "9606" & Taxid_interactor_B == "10090")  | 
                                (Taxid_interactor_A == "10090" & Taxid_interactor_B == "9606")) &
                               interactor_IDs_databases_A == "uniprotkb" & interactor_IDs_databases_B == "uniprotkb" &
                               ((bait_prey_status_A == "prey" & bait_prey_status_B == "bait")  | 
                                  (bait_prey_status_A == "bait" & bait_prey_status_B == "prey")) &
                               Host_organisms == "9606"
                             ,.(IDs_interactor_A, IDs_interactor_B, Taxid_interactor_A, Taxid_interactor_B)]
  mouse_as_bait_human_preys = length(unique(c(mouse_as_bait[(Taxid_interactor_A == "9606"), IDs_interactor_A], 
                                              mouse_as_bait[(Taxid_interactor_B == "9606"), IDs_interactor_B])))
  mouse_as_bait_mouse_baits = length(unique(c(mouse_as_bait[(Taxid_interactor_A == "10090"), IDs_interactor_A], 
                                              mouse_as_bait[(Taxid_interactor_B == "10090"), IDs_interactor_B])))
  N_pmid_mouse_as_bait_in_human = all.IntAct[((Taxid_interactor_A == "9606" & Taxid_interactor_B == "10090")  | 
                                                (Taxid_interactor_A == "10090" & Taxid_interactor_B == "9606")) &
                                               interactor_IDs_databases_A == "uniprotkb" & interactor_IDs_databases_B == "uniprotkb" &
                                               ((bait_prey_status_A == "prey" & bait_prey_status_B == "bait")  | 
                                                  (bait_prey_status_A == "bait" & bait_prey_status_B == "prey")) &
                                               Host_organisms == "9606"
                                             ,length(unique(Publication_Identifiers))]
  all.IntAct[Taxid_interactor_A == "9606" & Taxid_interactor_B == "9606", table(Host_organisms)]
}

```

Surprisingly, `r the_host_for_mouse_human_pairs[["9606"]]` interactions between mouse and human proteins were discovered in human rather than mouse cells (only `r the_host_for_mouse_human_pairs[["10090"]]`) suggesting that researchers use mouse rather than human proteins as baits (`r mouse_as_bait_mouse_baits` mouse baits total, `r mouse_as_bait_human_preys` human preys total, including isoforms, from `r N_pmid_mouse_as_bait_in_human` publications) to find interactions directly relevant to human interactome research, including human disease.  

```{r human_not_in_IMEx_protein_properties_processing}
# this code chunk adds protein mass data to interaction dataset used for Venn diagrams
SPECIES_NAME = c("Homo sapiens", "Mus musculus", "strain ATCC 204508", "strain K12")
## only reviewed = 2 and isoforms = FALSE is relevant for this analysis
reviewed = 2
isoforms = FALSE
##============================================================================##
source("SPECIES_NAME_TO_ID.R")
for (r in reviewed) {
  for (i in isoforms) {
    for (n in SPECIES_NAME) {
      ##============================================================================##
      ## querying Uniprot for the list of proteins (only mapped to Uniprot)
      ## downolading length, mass, SNPs, isoforms, annotation score, existence attribute
      SPECIES_IDs = SPECIES_NAME_TO_ID(n)
      SPECIES_ID = SPECIES_IDs$SPECIES_ID
      ## reading logic table and getting list of proteins
      if(n != "strain K12"){
        filename_vs_3 = paste("./analysis/","proteome_vs_interactome_vs_BioGRID_f_", SPECIES_ID,"_reviewed_",r,"_isoforms_",i,"_", date,".txt", sep = "")}
      if(n == "strain K12"){
        filename_vs_3 = paste("./analysis/","proteome_vs_interactome_f_", SPECIES_ID,"_reviewed_",r,"_isoforms_",i,"_", date,".txt", sep = "")}
      biogrid_from_mentha_vs_proteome_vs_imex_f = as.data.frame(read.delim(filename_vs_3, stringsAsFactors = F, quote = ""))
      source("download_whole_proteome.R")
      whole_proteome_query = download_whole_proteome(SPECIES_ID, date = date)
      colnames(whole_proteome_query)[2] = "Status"
      uniprot_query = dplyr::filter(whole_proteome_query, Status == "reviewed")
      ##============================================================================##
      ## merging information from Uniprot to the logic table
      proteome_vs_imex_details_f = merge(filter(biogrid_from_mentha_vs_proteome_vs_imex_f, whole_proteome_Uniprot==1), 
                                         uniprot_query, 
                                         by.x = "whole_proteome_IDs",
                                         by.y = "Entry")
      proteome_vs_imex_details_f$Mass = gsub(",","",proteome_vs_imex_details_f$Mass)
      proteome_vs_imex_details_f$Mass = as.numeric(proteome_vs_imex_details_f$Mass)
      ## creating a factor variable for presence_in_Uniprot.presence_in_IMEx (1_0/1_1)
      proteome_vs_imex_details_f[,length(proteome_vs_imex_details_f)+1] = interaction(proteome_vs_imex_details_f$whole_proteome_Uniprot, proteome_vs_imex_details_f$IMEx, sep = "_")
      colnames(proteome_vs_imex_details_f)[length(proteome_vs_imex_details_f)] = paste0(colnames(proteome_vs_imex_details_f)[2], "_","IMEx")
      levels(proteome_vs_imex_details_f$whole_proteome_Uniprot_IMEx) = c("SwissProt_not_IMEX", "SwissProt_and_IMEX")
      ##============================================================================##
      ## saving combined logic table + protein properties from Uniprot
      filename_vs_2 = paste("./analysis/","proteome_vs_interactome_protein_properties_f_", n,"_reviewed_",r,"_isoforms_",i,"_", date,".txt", sep = "")
      write.table(proteome_vs_imex_details_f,filename_vs_2,col.names=T,row.names=F,sep="\t",quote=F)
    }
  }
}
```

### GO enrichment: molecular function: Olfactory receptors are a major group of human proteins not represented in IntAct

We have performed Gene Ontology enrichment analysis of proteins with no available interactions as compared to whole proteome (SwissProt protein list). We have looked at overrepresented biological processes, molecular fuctions and cellular compartment localisation of proteins. Here only overrepresented biological processes are shown.  

```{r human_not_in_IMEx_protein_properties_GO_BP, fig.width=10, fig.height=8, fig.cap="Overrepresented biological processes. The size of the dot shows the number of proteins of that category in the dataset (proteins without interaction), x-axis represents the fraction of proteins of that category in the dataset. Colour shows the probability of that overlap occurring by chance."}
#### This chunk does GO enrichment
##====================================================================
suppressPackageStartupMessages({
  library(data.table)
  library(qvalue)
  library(ggplot2)
})
##====================================================================
# read the saved table
SPECIES_NAME = "Homo sapiens"
n = SPECIES_NAME
## only reviewed = 2 is relevant for this analysis
reviewed = 2
r = reviewed
isoforms = FALSE
i = isoforms
filename_vs_2 = paste("./analysis/","proteome_vs_interactome_protein_properties_f_", n,"_reviewed_",r,"_isoforms_",i,"_", date,".txt", sep = "")
proteome_vs_imex_details_f = fread(filename_vs_2, header = T, stringsAsFactors = F,quote="")
##====================================================================

not_in_IMEx_BP = GO_enrich_simplify_plot_bioc(protein_set = unique(proteome_vs_imex_details_f[whole_proteome_Uniprot == 1 & IMEx == 0,whole_proteome_IDs]),
                                              reference_protein_set = unique(proteome_vs_imex_details_f[whole_proteome_Uniprot == 1,whole_proteome_IDs]),
                                              identifier_type = "UNIPROT", ontology = "BP", 
                                              pAdjustMethod_ = "BH", 
                                              minSetSize = 40, maxSetSize = 430, 
                                              simplify_by = "GeneRatio", 
                                              simplify_fun = "function (x) x[which.max(eval(parse(text = paste0(\"c(\",paste0(x, collapse = \",\"),\")\"))))]", 
                                              similarity_calc_method = "Wang", 
                                              similarity_cutoff = 0.8, 
                                              visualize_result = "dotplot", 
                                              above_corrected_pval = 0.05, 
                                              use_bioc_annotationdbi = T, 
                                              plot_title = "biological function of proteins with no interactions in IntAct",
                                              xlabel = "the fraction of proteins with a specific biological function \namong proteins with no interactions",
                                              drop_GO_levels = c(1:3,15:20))
print(not_in_IMEx_BP$plot + theme(plot.title = element_text(hjust = 0.8)))
```

### Human proteins with no available interactions are on average shorter than the proteins with interactions available  

```{r human_not_in_IMEx_proteins_are_shorter_density, fig.width=7, fig.height=5, fig.cap="Human proteins with no available interactions are on average shorter than the proteins with interactions available. The plot compares the distribution of protein mass across SwissProt proteins with or without interacting partners in IntAct. Vertical line highlights the median." }
#### This chunk prepares the data to plot the Bias in mass using ggplot
## Analysis
# read the saved table
SPECIES_NAME = "Homo sapiens"
n = SPECIES_NAME
## only reviewed = 2 is relevant for this analysis
reviewed = 2
r = reviewed
isoforms = FALSE
i = isoforms
filename_vs_2 = paste("./analysis/","proteome_vs_interactome_protein_properties_f_", n,"_reviewed_",r,"_isoforms_",i,"_", date,".txt", sep = "")
proteome_vs_imex_details_f = as.data.frame(read.delim(filename_vs_2, header = T, stringsAsFactors = F, quote=""))
proteome_vs_imex_details_f$whole_proteome_Uniprot_IMEx = factor(proteome_vs_imex_details_f$whole_proteome_Uniprot_IMEx, ordered =F)
## plotting the distribution of mass
# ggplot(proteome_vs_imex_details_f, aes(x = Mass, color = whole_proteome_Uniprot_IMEx, alpha =0.5)) +geom_density()+ scale_x_log10()
# ggplot(proteome_vs_imex_details_f, aes(x = Mass, color = whole_proteome_Uniprot_IMEx, alpha =0.5)) + scale_x_log10() + geom_histogram(position = "identity", bins = 50)

## removing olfactory receptors:
{proteome_vs_imex_details_f_minus_odor = proteome_vs_imex_details_f[-grep("Odor", 
                                                                          proteome_vs_imex_details_f$Protein.names),]
  proteome_vs_imex_details_f_minus_odor_olf = proteome_vs_imex_details_f_minus_odor[-grep("Olfactory",
                                                                                          proteome_vs_imex_details_f_minus_odor$Protein.names),]
  proteome_vs_imex_details_f_minus_odor_olf.c = rbind(cbind(proteome_vs_imex_details_f, olfac = 
                                                              rep("all SwissProt", nrow(proteome_vs_imex_details_f))),
                                                      cbind(proteome_vs_imex_details_f_minus_odor_olf, olfac = rep("SwissProt excluding \n olfactory receptors",nrow(proteome_vs_imex_details_f_minus_odor_olf))))}

# the number of odor receptors excluded
n_odor_receptors = length(unique(grep("Odor", proteome_vs_imex_details_f$Protein.names, value = F))) + length(unique(grep("Olfactory", proteome_vs_imex_details_f_minus_odor$Protein.names, value = F)))

# calculating median per group
proteome_vs_imex_details_f_minus_odor_olf.c = as.data.table(proteome_vs_imex_details_f_minus_odor_olf.c)
proteome_vs_imex_details_f_minus_odor_olf.c[, whole_proteome_Uniprot_IMEx_x_olfac := paste0(whole_proteome_Uniprot_IMEx,olfac)]
proteome_vs_imex_details_f_minus_odor_olf.c[, median_Mass_per_group := median(log10(Mass)), by = whole_proteome_Uniprot_IMEx_x_olfac]

proteome_vs_imex_details_f_minus_odor_olf.c[, in_IntAct := ifelse(whole_proteome_Uniprot_IMEx == "SwissProt_and_IMEX", "yes", "no")]
```

```{r distribution_of_mass}
#### This chunk plots the Bias in mass using ggplot
## histogram with and without olfactory receptors:
ggplot(proteome_vs_imex_details_f_minus_odor_olf.c, aes(x = Mass, color = in_IntAct, label = olfac)) +geom_freqpoly(bins = 100)+ scale_x_log10() + theme(strip.text.y = element_text(angle = 0)) +xlab("protein mass, Da, log10 scale") + ggtitle("The distribution of protein mass for proteins with and \n without interacting partners curated in IMEx")+geom_vline(aes(xintercept = 10^median_Mass_per_group, color = in_IntAct),alpha =0.7)  + facet_grid(olfac~.)+
  geom_vline(aes(xintercept = 10^median_Mass_per_group, color = in_IntAct),alpha =0.7,size =1.5) +
  theme(panel.grid.major =  element_line(color = 'grey', size = 0.2, linetype = 'solid'),
        panel.background = element_rect(fill = '#FFFFFF', colour = 'grey')) + 
  scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x),
                labels = trans_format("log10", math_format(10^.x)))
```

### Wilcox test for difference in distribution means 

```{r human_not_in_IMEx_proteins_are_shorter_Wilcox}
## testing the difference in protein mass distributions
##of proteins absent/present in IMEx using Wilcox rank test
SwissProt_not_IMEX = proteome_vs_imex_details_f$Mass[proteome_vs_imex_details_f$whole_proteome_Uniprot_IMEx=="SwissProt_not_IMEX"]
SwissProt_and_IMEX = proteome_vs_imex_details_f$Mass[proteome_vs_imex_details_f$whole_proteome_Uniprot_IMEx=="SwissProt_and_IMEX"]
wilcox.test.r = wilcox.test(SwissProt_not_IMEX, SwissProt_and_IMEX, conf.int = T)
ttest_r = t.test(log10(SwissProt_not_IMEX), log10(SwissProt_and_IMEX), conf.int = T)
## testing the difference in protein mass distributions after removing odor receptors
##of proteins absent/present in IMEx using Wilcox rank test excluding olfactory receptors
SwissProt_not_IMEX = proteome_vs_imex_details_f_minus_odor_olf$Mass[proteome_vs_imex_details_f_minus_odor_olf$whole_proteome_Uniprot_IMEx=="SwissProt_not_IMEX"]
SwissProt_and_IMEX = proteome_vs_imex_details_f_minus_odor_olf$Mass[proteome_vs_imex_details_f_minus_odor_olf$whole_proteome_Uniprot_IMEx=="SwissProt_and_IMEX"]
wilcox.test.r.min.olf = wilcox.test(SwissProt_not_IMEX, SwissProt_and_IMEX, conf.int = T)
```

This difference in protein mass between proteins present and absent in the interactome is highly unlikely to occur by chance (Wilcox rank test (Mass, Da, 95% confidence interval: `r format(signif(wilcox.test.r$conf.int,3), scientific =F)`, p-value: `r format(signif(wilcox.test.r$p.value,3), scientific =F)`) and Student t-test(log10 of Mass, Da, 95% confidence interval: `r format(signif(ttest_r$conf.int,3), scientific =F)`, p-value: `r format(signif(ttest_r$p.value,3), scientific =F)`) on the whole population of proteins. The removal of `r format(signif(n_odor_receptors,3), scientific =F)` olfactory receptors, evidently, does not change this trend (Wilcox rank test on Mass, Da, 95% confidence interval: `r format(signif(wilcox.test.r.min.olf$conf.int,3), scientific =F)`, p-value: `r format(signif(wilcox.test.r.min.olf$p.value,3), scientific =F)`).  

## ##3. Bar plot for interspecies interactions: BioGrid from Mentha

```{r Supplementary2_BioGRID_vs_IMEx_vs_Uniprot_N_Uniprot_Species, cache=F, fig.width=8, fig.height=8, fig.cap="Supplementary figure 2. The number of interacting proteins from given species or the other species in BioGRID. Red bar shows the number of proteins from a given species, blue bar shows the number of proteins from another species. Red group proteins form interactions both with red and blue group proteins, blue group proteins form interactions only with red group proteins." }
#### This chunk plots using ggplot
### combine the summaries for multiple species of how many interactors have non-uniprot identifiers
interactome_identifiers_summary.all = data.frame()
for (i in isoforms) {
  for(n in SPECIES_NAME) {
    filename.summary = paste("./summaries/","uniprotKB_IDs_and_",n,"_biogrid_from_mentha_interactors_summary", "_isoforms_",i,"_", date,".txt", sep = "")
    if(!file.exists(filename.summary)) {
      interactome_identifiers_summary.all
    }
    if(file.exists(filename.summary)) {
      interactome_identifiers_summary.o = as.data.frame(read.delim(filename.summary, stringsAsFactors = F, quote = ""))
      interactome_identifiers_summary.all = rbind(interactome_identifiers_summary.all, interactome_identifiers_summary.o)
    }
  }
}
interactome_identifiers_summary.all = unique(interactome_identifiers_summary.all)
## save all species summary
filename.summary.all = paste("./results/","interactome_identifiers_BioGRID_summary_",date,".txt", sep = "")
write.table(interactome_identifiers_summary.all, filename.summary.all, col.names=T,row.names=F,sep="\t",quote=F)

#################
interactome_identifiers_summary.all.s = interactome_identifiers_summary.all[,c(3,4,5,6,7)]
## Transform table for plotting with ggplot2
interactome_identifiers_summary.all.s.m= melt(data = interactome_identifiers_summary.all.s,
                                              id.vars = c("SPECIES_NAME", "SPECIES_ID"),
                                              variable.name = "decription",
                                              value.name = "number")
interactome_identifiers_summary.all.s.m=unique(interactome_identifiers_summary.all.s.m)
## rename yeast and E.coli
interactome_identifiers_summary.all.s.m$SPECIES_NAME = gsub("strain ATCC 204508", "S. cerevisiae, strain S288c", interactome_identifiers_summary.all.s.m$SPECIES_NAME)
interactome_identifiers_summary.all.s.m$SPECIES_NAME = gsub("strain K12", "E. coli, strain K12", interactome_identifiers_summary.all.s.m$SPECIES_NAME)
interactome_identifiers_summary.all.s.m$SPECIES_NAME = gsub("Caenorhabditis elegans", "C. elegans", interactome_identifiers_summary.all.s.m$SPECIES_NAME)
interactome_identifiers_summary.all.s.m$SPECIES_NAME = gsub("Drosophila melanogaster", "D. melanogaster", interactome_identifiers_summary.all.s.m$SPECIES_NAME)

# filter out non-uniprot identifiers to make picture clearer
interactome_identifiers_summary.all.s.m$decription = gsub("\\.", " ", interactome_identifiers_summary.all.s.m$decription)
ind = interactome_identifiers_summary.all.s.m$decription != "interacting partners with the other identifier"
interactome_identifiers_summary.all.s.m = interactome_identifiers_summary.all.s.m[ind,]

# plot
proteome_vs_interactome_plot <- ggplot(interactome_identifiers_summary.all.s.m, aes(x=SPECIES_NAME, y=number, fill=decription,label=number)) + geom_bar(width = 0.9, stat = "identity", position = "stack") + geom_label(position = "stack", size = 4, label.padding = unit(0.08, "lines")) +
  ggtitle("The number of interacting partners which have UniprotKB identifiers \n and belong to the species of interest") + theme(axis.text.x=element_text(angle = 20), axis.title.x = element_text(size = 0)) + ylab("number of proteins")
proteome_vs_interactome_plot
# save plot
# filename=paste("./results/", "interactome_identifiers_summary_plot_SMALL",date,".png", sep = "")
# ggsave(filename, proteome_vs_interactome_plot, width = 12, height = 12)

### ======================================================================== ###

```